internal/inventory/app/routes.go

```diff
@@ -59,90 +60,95 @@ func (app *InventoryApp) addRoutes() error {
-	http.ServeFile(w, req, "static/html/index.html")
+func (app *InventoryApp) index(ctx echo.Context) error {
+	http.ServeFile(ctx.Response().Writer, ctx.Request(), "static/html/index.html")
+	return nil
 }
 
-func (app *InventoryApp) getReaders(w http.ResponseWriter, _ *http.Request) {
+func (app *InventoryApp) getReaders(ctx echo.Context) error {
+	w := ctx.Response().Writer
 	w.Header().Set("Content-Type", "application/json")
 	if err := app.defaultGrp.WriteReaders(w); err != nil {
 		msg := fmt.Sprintf("Failed to write readers list: %v", err)
 		app.lc.Error(msg)
-		w.WriteHeader(http.StatusInternalServerError)
-		http.Error(w, msg, http.StatusInternalServerError)
+		ctx.String(http.StatusInternalServerError, msg)
 	}
+	return nil
 }
 
-func (app *InventoryApp) getSnapshot(w http.ResponseWriter, _ *http.Request) {
+func (app *InventoryApp) getSnapshot(ctx echo.Context) error {
+	w := ctx.Response().Writer
 	w.Header().Set("Content-Type", "application/json")
 	if err := app.requestInventorySnapshot(w); err != nil {
 		msg := fmt.Sprintf("Failed to write inventory snapshot: %v", err)
 		app.lc.Error(msg)
-		w.WriteHeader(http.StatusInternalServerError)
-		http.Error(w, msg, http.StatusInternalServerError)
+		ctx.String(http.StatusInternalServerError, msg)
 	}
+	return nil
 }
 
-func (app *InventoryApp) startReading(w http.ResponseWriter, _ *http.Request) {
+func (app *InventoryApp) startReading(ctx echo.Context) error {
 	if err := app.defaultGrp.StartAll(app.devService); err != nil {
 		msg := fmt.Sprintf("Failed to StartAll: %v", err)
 		app.lc.Error(msg)
-		w.WriteHeader(http.StatusInternalServerError)
-		http.Error(w, msg, http.StatusInternalServerError)
+		ctx.String(http.StatusInternalServerError, msg)
 	}
+	return nil
 }
 
-func (app *InventoryApp) stopReading(w http.ResponseWriter, _ *http.Request) {
+func (app *InventoryApp) stopReading(ctx echo.Context) error {
 	if err := app.defaultGrp.StopAll(app.devService); err != nil {
 		msg := fmt.Sprintf("Failed to StopAll: %v", err)
 		app.lc.Error(msg)
-		w.WriteHeader(http.StatusInternalServerError)
-		http.Error(w, msg, http.StatusInternalServerError)
-		return
+		ctx.String(http.StatusInternalServerError, msg)
 	}
+	return nil
 }
 
-func (app *InventoryApp) getBehavior(w http.ResponseWriter, req *http.Request) {
-	rv := mux.Vars(req)
+func (app *InventoryApp) getBehavior(ctx echo.Context) error {
+	w := ctx.Response().Writer
+	rv := mux.Vars(ctx.Request())
 	bName := rv["name"]
 	// Currently, only "default" is supported.
 	if bName != "default" {
 		msg := fmt.Sprintf("Request to GET unknown behavior. Name: %v", bName)
 		app.lc.Error(msg)
+
 		if _, err := w.Write([]byte("Invalid behavior name.")); err != nil {
 			app.lc.Error("Error writing failure response.", "error", err)
 		}
-		w.WriteHeader(http.StatusNotFound)
-		http.Error(w, msg, http.StatusNotFound)
-		return
+		ctx.String(http.StatusNotFound, msg)
+		return nil
```

Comment:
Make same change to similar code below
```suggestion
		return ctx.String(http.StatusNotFound, msg)
```